src/contracts/ProofOfPhysicalAddress.contract.js

```diff
@@ -15,42 +15,41 @@ export default class ProofOfPhysicalAddress {
     let proofOfPhysicalAddressAbi = await helpers.getABI(branch, 'ProofOfPhysicalAddress')
     this.instance = new web3_10.eth.Contract(proofOfPhysicalAddressAbi, PROOF_OF_PHYSICAL_ADDRESS)
 
-    this.getPhysicalAddressesOfWalletAddressArray = this.getPhysicalAddressesOfWalletAddressArray.bind(this)
+    this.getPhysicalAddressesOfWalletAddress = this.getPhysicalAddressesOfWalletAddress.bind(this)
     this.getAllEvents = this.getAllEvents.bind(this)
+    this.getAllAddressRegisteredEvents = this.getAllAddressRegisteredEvents.bind(this)
     this.getPhysicalAddressesByWalletAddressAndKeccakIdentifierArray = this.getPhysicalAddressesByWalletAddressAndKeccakIdentifierArray.bind(
       this
     )
   }
 
   /**
-   * Given a wallet address array, return a promise that resolves to an array of physical addresses from
-   * PoPA contract.
-   * @param {String[]} walletAddressArray
+   * Given a wallet address, return a promise that resolves to an array of physical addresses from PoPA contract or null.
+   * If a cached registeredAddressEvents is provided, it will be used to get the corresponding keccakIdentifiers from it.
```

Comment:
Changed the method to fetch the physical addresses of an individual _wallet_ address, instead of an array of _wallet_ addresses. This implies that the `registeredAddressEvents` needs to be passed to the function (if ommited, the function will get them).

`AllValidators` does this events retrieval once and keeps the result on a variable to avoid making extra queries/requests.